src/python/__init__.py

```diff
@@ -0,0 +1,485 @@
+                The name of node to execute the operation
+
+            unitName:
+                Unit Name to be disabled
+
+        Returns:
+        """
+        node_path = self.manager.GetNode(nodeName)
+        node = self.bus.get_proxy(
+            HIRTE_DBUS_INTERFACE,
+            node_path
+        )
+
+        node.DisableUnitFiles(
+            unitName,
+            False,
+        )
+
+    def _enableUnit(
+        self,
+        nodeName,
+        unitName
+    ):
+        """
+        _enableUnit() implements the Enable operation
+        Args:
+            nodeName:
+                The name of node to execute the operation
+
+            unitName:
+                Unit Name to be enabled
+
+        Returns:
+        """
+
+        EnabledServiceInfo = namedtuple(
+            "EnabledServicesInfo",
+            ["op_type", "symlink_file", "symlink_dest"]
+        )
+
+        EnableResponse = namedtuple(
+            "EnableResponse",
+            ["carries_install_info", "enabled_services_info"]
+        )
+
+        node_path = self.manager.GetNode(nodeName)
+        node = self.bus.get_proxy(
+            HIRTE_DBUS_INTERFACE,
+            node_path
+        )
+
+        response = node.EnableUnitFiles(
+            unitName,
+            False,
+            False
+        )
+
+        enable_response = EnableResponse(*response)
+        if enable_response.carries_install_info:
+            print("The unit files included enablement information")
+        else:
+            print("The unit files did not include any enablement information")
+
+        for e in enable_response.enabled_services_info:
+            enabled_service_info = EnabledServiceInfo(*e)
+            if enabled_service_info.op_type == "symlink":
+                print(f"Created symlink {enabled_service_info.symlink_file}"
+                      " -> {enabled_service_info.symlink_dest}")
+            elif enabled_service_info.op_type == "unlink":
+                print(f"Removed \"{enabled_service_info.symlink_file}\".")
+
+    def ListAllNodes(
+        self,
+    ):
+        """
+        The public method ListAllNodes() will print all Hirte nodes.
+
+        Returns:
+        """
+        NodeInfo = namedtuple(
+            "NodeInfo", [
+                "name",
+                "object_path",
+                "status"]
+        )
+
+        nodes = self.manager.ListNodes()
+        for node in nodes:
+            info = NodeInfo(*node)
+            print(f"Node: {info.name}, State: {info.status}")
+
+    def SetCPUWeight(
+        self,
+        nodeName,
+        unitName,
+        value,
+        persist=False
+    ):
+        """
+        Set CPU Weight
+
+        Args:
+            nodeName:
+                The name of node to execute the operation
+
+            unitName:
+                Unit Name
+
+            value:
+                value to be set
+
+            persist:
+                Set True to NOT persist de change
```

Comment:
This is confusing, should we invert the logic? We could simply do below : `runtime = not persist` and then the semantics is respected.

Reply:
I just removed persist and left runtime. In case you would like to change the runtime behavior we will need a change in the hirte daemon.